test/metabase/query_processor/middleware/fix_bad_references_test.clj

```diff
@@ -0,0 +1,32 @@
+(ns metabase.query-processor.middleware.fix-bad-references-test
+  (:require [clojure.test :refer :all]
+            [metabase.query-processor.middleware.fix-bad-references :as fix-bad-refs]
+            [metabase.test :as mt]))
+
+(defn- fix-bad-refs [query]
+  (mt/with-everything-store
+    (fix-bad-refs/fix-bad-references query)))
+
```

Comment:
I couldn't hit the exception, just the warning - maybe worthwhile to explicitly hit that exception, because if the underlying goal is to whack the undefined behavior, defaulting into undefined behavior after failing at the guess would not be happy times

Reply:
Yeah that was my original strategy but unfortunately whacking the undefined behavior here would mean fixing broken behavior on the FE client and fixing existing broken queries which is easier said than done -- so I reworked this middleware to be more of a DWIM best-effort thing

Reply:
I mean clearly it is _possible_ for those broken queries to work somehow, so theoretically it should be possible to transform the broken queries into a correct query that returns the same results, but I didn't go that far down that rabbit hole TBH. I think that would require a much more complicated namespace than this new one and since this wasn't really the main focus of #19384 I stopped with the simple transformation that fixed about 95% of the problems